crates/sui-types/src/crypto.rs

```diff
@@ -273,6 +274,14 @@ impl Signature {
         secret.sign(&message)
     }
 
+    /// TODO: This method will replace `new` in an upcoming PR.
+    pub fn new_secure<T>(value: &T, secret: &dyn signature::Signer<Signature>) -> Self
+    where
+        T: SecureIntent,
+    {
+        secret.sign(&bcs::to_bytes(&value).expect("Message serialization should not fail"))
+    }
+
```

Comment:
Why not implement an [Extension trait](http://xion.io/post/code/rust-extension-traits.html) for `Signer`?

Reply:
Good point, I just got rid of the `Signable.write` which still serialized + added the domain separation prefix (and now we don't need it).
```
let mut message = Vec::new();
        value.write(&mut message);
        secret.sign(&message)
```